cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -75,6 +105,145 @@ public CouchClient(URI rootUri,
+                String tlsv1_2 = "TLSv1.2";
+                // Get the API level reflectively so we don't need to import classes only available
+                // in Android
+                int androidApiLevel = Class.forName("android.os.Build$VERSION").getField
+                        ("SDK_INT").getInt(null);
+                // If we are on an old version of Android and TLSv1.2 has not been enabled already
+                // on the default context then we add a special interceptor
+                if (androidApiLevel < 20) {
+                    // Check i has passed we are on an old version of Android
+
+                    // Get the default SSLContext
+                    SSLContext defSslCtx = SSLContext.getDefault();
+
+                    // Check the default protocols for TLSv1.2
+                    if (!Arrays.asList(defSslCtx.getDefaultSSLParameters().getProtocols())
+                            .contains(tlsv1_2) &&
+                            Arrays.asList(defSslCtx.getSupportedSSLParameters().getProtocols())
+                                    .contains(tlsv1_2)) {
+                        // Check ii has passed TLSv1.2 is not already enabled on the default context
+                        // but is supported on the default context so we can use it
+
+                        // In an ideal world we would also check the default SSLSocketFactory that
+                        // is set on the HttpsUrlConnection to see if that has been customized from
+                        // the default, but there isn't really a way to check this, equals is not
+                        // overridden on the SSLSocketFactory and there is no route back to the
+                        // SSLContext that generated the SSLSocketFactory.
+                        SSLContext sc = SSLContext.getInstance(tlsv1_2);
+                        sc.init(null, null, null);
+                        // We add this interceptor, but it could be overridden by a later user
+                        // provided interceptor, so this shouldn't change anyone's existing
+                        // customizations.
+                        SSLSocketFactory s = sc.getSocketFactory();
+                        return new Special(s);
+                    }
+                }
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException(e);
+            } catch (NoSuchFieldException e) {
+                throw new RuntimeException(e);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException(e);
+            } catch (KeyManagementException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return null;
+    }
+
+    private static final class TlsOnlySslSocketFactory extends SSLSocketFactory {
+
+        private static final String[] TLSv12Only = new String[]{"TLSv1.2"};
```

Comment:
This string also occurs in `checkAndGetTlsInterceptor`, maybe have a static const somewhere?

Reply:
Done in 9d877ca53d472c12993fada59740b3c77880b226